internal/pkg/skuba/kubernetes/pod.go

```diff
@@ -30,8 +32,23 @@ func getPodContainerImageTag(client clientset.Interface, namespace string, podNa
 	if err != nil {
 		return "", errors.Wrap(err, "could not retrieve pod object")
 	}
+	return getPodContainerImageTagFromPodObject(podObject), nil
+}
+
+// return the image version tag without calling the api, just with the Pod object
+func getPodContainerImageTagFromPodObject(podObject *v1.Pod) string {
 	containerImageWithName := podObject.Spec.Containers[0].Image
 	containerImageTag := strings.Split(containerImageWithName, ":")
 
-	return containerImageTag[len(containerImageTag)-1], nil
+	return containerImageTag[len(containerImageTag)-1]
+}
+
+// return a pod object matched from a podList
+func getPodFromPodList(list *v1.PodList, name string) (*v1.Pod, error) {
```

Comment:
A minor improvement in terms of speed would be to construct a map of `map[string]*v1.Pod` (or even `map[string]string`) with the pod name as key (in the second case we would have the image as the value). That way we iterate only once, constructing this map, and read from the map.

Reply:
I thought about that. Will it be faster? does adding that complexity and unreadability in code outweighs the speed gain? (I guess we should test that with max pods (250) to see the speed change.

Reply:
you have an example with the nodes info, they are returning a map
https://github.com/SUSE/skuba/pull/897/files#diff-9236f8f0e4ba056ee3aa9030db67f0a0R125-R142

find an element in a map O(1) vs O(N) using a for loop :)

Reply:
that doesnt really answer my question of introducing complexity vs speed gain :p

Maybe is not that complex and it only seems to be because Im a go newbie, you know as a pythonista we say "its fast enough" :P

Reply:
I think that both approaches should be fine for this case, just a bit of bikeshedding on Mondays doesn't hurt :)

Reply:
I agree that it may not be worth at this time (maybe O(n) is faster with this small amount than lookups O(1) and insertions O(n*log(n))? However if the number of pods grow on `kube-system` we will probably hit where both complexities cross and a map is always more efficient after that crossing.

I'm fine with your current solution though.

Reply:
To be fair, Im not against that, but I rather move this forward to then introduce the retry part ASAP as that fixes a bug, *then* rework this by adding the map-ing to squeeze more speed out of it

Reply:
The default number of pods in kube-system looks like only 20:

```
$ kubectl --kubeconfig ./admin.conf get pods -n kube-system | wc -l
20
```
which is basically nothing. Is it likely to grow many more pods? Even if it was 100 I don't think that's so much to be worth comparing O(1) to O(N).

Reply:
We don't know to what can grow (users can deploy whatever they prefer on this namespace), but I agree there's no real benefit in thinking about optimizing this at this point (if we have evidence that there's an issue we can take action, but it's perfectly fine for now IMO).